tfhe/src/high_level_api/compact_list.rs

```diff
@@ -31,6 +82,20 @@ impl CompactCiphertextList {
         CompactCiphertextListBuilder::new(pk)
     }
 
+    pub fn len(&self) -> usize {
+        self.0.len()
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.len() == 0
+    }
+
+    pub fn get_kind_of(&self, index: usize) -> Option<crate::FheTypes> {
+        self.0.get_kind_of(index).and_then(|data_kind| {
+            crate::FheTypes::from_data_kind(data_kind, self.0.ct_list.message_modulus)
+        })
+    }
```

Comment:
does not work for packed lists ? the modulus would be message_modulus * carry_modulus ?

Reply:
Yes, it work because the metada is metadata, it does not care whether the actual block are packed or not

The integer metadata keeps string of the bool or signed/unsigned kind with the number of blocks or the original input, the hlapi list just require the message modulus to be able to know how many bits a block encrypts to then be able to translate a num blocks in a num bits